### PR TITLE
Add Prometheus Remote Write Exporter (1/6)

### PR DESCRIPTION
**Pull request by shovnik**
# Description
This is PR 1/6 of adding a Prometheus Remote Write Exporter in Python SDK and address Issue https://github.com/open-telemetry/opentelemetry-python/issues/1302

👉 **[Part 1/6](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/180)**
* Adds class skeleton
* Adds all function signatures

**[Part 2/6](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/206)**
* Adds validation of exporter constructor commands
* Add unit tests for validation

**[Part 3/6](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/207)**
* Adds conversion methods from OTel metric types to Prometheus [TimeSeries](https://github.com/prometheus/prometheus/blob/master/prompb/types.proto#L47)
* Add unit tests for conversion

**[Part 4/6](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/212)**
* Adds methods to export metrics to Remote Write endpoint
* Add unit tests for exporting

**[Part 5/6](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/216)**
* Add integration tests using sample app and instance of Cortex

**Part 6/6**
* Add README, Design Doc and other necessary documentation.

## Type of change
- [x] New feature (non-breaking change which adds functionality)
# How Has This Been Tested?

- [ ] Added class `TestValidation` in `test_prometheus_remote_write_exporter.py`

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:
- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [ ] Documentation has been updated (Full documentation in final exporter PR)
cc- @AzfaarQureshi , @alolita

### CODE REVIEW

**Comment by lzchen** [comment]
Is this ready for review?

**Comment by shovnik** [comment]
Yes, this PR is now ready for review (updated description). The three next PRs mentioned in the description should be up for review by end of day as well. @lzchen

**Comment by AzfaarQureshi** [comment]
@lzchen the PR is kind of big because of computer generated Protobufs and copy-pasted licenses. The actual code is within `__init__.py` and the test folder

File: exporter/opentelemetry-exporter-prometheus-remote-write/README.rst
Comment (codeboten) [inline exporter/opentelemetry-exporter-prometheus-remote-write/README.rst:22]:
```suggestion
* `Prometheus <https://prometheus.io/>`_
```

File: exporter/opentelemetry-exporter-prometheus-remote-write/README.rst
Comment (codeboten) [inline exporter/opentelemetry-exporter-prometheus-remote-write/README.rst:4]: Should this say Prometheus Remote Write?

File: exporter/opentelemetry-exporter-prometheus-remote-write/setup.cfg
Comment (codeboten) [inline exporter/opentelemetry-exporter-prometheus-remote-write/setup.cfg:22]: url needs updating

File: exporter/opentelemetry-exporter-prometheus-remote-write/setup.cfg
Comment (codeboten) [inline exporter/opentelemetry-exporter-prometheus-remote-write/setup.cfg:42]: is ddtrace necessary?

**Review by codeboten — CHANGES_REQUESTED** [review]
A few minor changes needed, would be good to add some info regarding the source of the protos for prometheus as well.

File: exporter/opentelemetry-exporter-prometheus-remote-write/CHANGELOG.md
Comment (lzchen) [inline exporter/opentelemetry-exporter-prometheus-remote-write/CHANGELOG.md:3]: I would include "initial release" and the PRs associated.

**Comment by lzchen** [comment]
Did we decide to create this in the `contrib` repo? I thought we wanted to keep this in the same place as the [prometheus](https://github.com/open-telemetry/opentelemetry-python/tree/master/exporter/opentelemetry-exporter-prometheus) exporter to avoid confusion?

**Comment by shovnik** [comment]
We were not sure where the exporter would reside so thought we would make the PRs on the contrib repo for now to get the reviews going. Once the standard location for prometheus exporters is decided, we can move one or the other.

File: exporter/opentelemetry-exporter-prometheus-remote-write/README.rst
Comment (shovnik) [reply exporter/opentelemetry-exporter-prometheus-remote-write/README.rst]: Updated

File: exporter/opentelemetry-exporter-prometheus-remote-write/README.rst
Comment (shovnik) [reply exporter/opentelemetry-exporter-prometheus-remote-write/README.rst]: Updated

File: exporter/opentelemetry-exporter-prometheus-remote-write/setup.cfg
Comment (shovnik) [reply exporter/opentelemetry-exporter-prometheus-remote-write/setup.cfg]: Updated

File: exporter/opentelemetry-exporter-prometheus-remote-write/setup.cfg
Comment (shovnik) [reply exporter/opentelemetry-exporter-prometheus-remote-write/setup.cfg]: No, it is a remnant from the datadog exporter I used as template, removed.

File: exporter/opentelemetry-exporter-prometheus-remote-write/CHANGELOG.md
Comment (shovnik) [reply exporter/opentelemetry-exporter-prometheus-remote-write/CHANGELOG.md:3]: Added Initial Release in CHANGELOG and will update it with each subsequent PR.

**Comment by shovnik** [comment]
I addressed all the requested changes @lzchen @codeboten

**Comment by toumorokoshi** [comment]
@lzchen @codeboten from a high level, how do you want to handle reviewing this six-parter?

It may be more efficient for the same people to review all PRs. happy to volunteer.

Currently I'm not too inclined to review 3/6 which I was assigned, since I have to go read 1 and 2 first anyway.

File: exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/gen/remote.proto
Comment (toumorokoshi) [inline exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/gen/remote.proto:1]: it looks like these are copied from the prometheus library itself. Probably fine. 

Are there any thoughts around what the update story would look like? just copy the protos over I suppose.

**Comment by lzchen** [comment]
@toumorokoshi 

I'm going to be reviewing all 6. Others are more than welcome to chime in as well.

**Review by lzchen — APPROVED** [review]

File: exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/gen/remote.proto
Comment (AzfaarQureshi) [reply exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/gen/remote.proto:1]: >  just copy the protos over I suppose.
Yup! thats the plan currently

File: exporter/opentelemetry-exporter-prometheus-remote-write/CHANGELOG.md
Comment (codeboten) [inline exporter/opentelemetry-exporter-prometheus-remote-write/CHANGELOG.md:5]: nit:
this change can just be under the `## Unreleased` section, when the release is done, the correct version tag title will be applied to the changelog

**Review by codeboten — COMMENTED** [review]
Changes look good, holding off on merging until we get this release out to prevent releasing the prometheus package before the other PRs are merged

**Review by ocelotl — APPROVED** [review]

File: exporter/opentelemetry-exporter-prometheus-remote-write/CHANGELOG.md
Comment (shovnik) [reply exporter/opentelemetry-exporter-prometheus-remote-write/CHANGELOG.md]: Rebased and updated CHANGELOG to use Unreleased

**Review by codeboten — APPROVED** [review]
Please update the branch and we can merge!

**Pull request merged**
